osbenchmark/worker_coordinator/worker_coordinator.py

```diff
@@ -198,6 +198,171 @@ def __init__(self, metrics, next_task_scheduled_in):
         self.metrics = metrics
         self.next_task_scheduled_in = next_task_scheduled_in
 
+class ClusterErrorMessage:
+    """Message sent from the client when a request fails during load testing"""
+    def __init__(self, client_id, request_metadata):
+        self.client_id = client_id
+        self.request_metadata = request_metadata
+
+class SharedClientStateMessage:
+    """Message sent from the Worker to the FeedbackActor to share client state dictionaries"""
+    def __init__(self, worker_id, dictionary):
+        self.worker_id = worker_id
+        self.worker_clients_map = dictionary
+
+class FeedbackState(Enum):
+    """Various states for the FeedbackActor"""
+    NEUTRAL = "neutral"
+    SCALING_DOWN = "scaling_down"
+    SLEEP = "sleep"
+    SCALING_UP = "scaling_up"
+
+class FeedbackActor(actor.BenchmarkActor):
+    POST_SCALEDOWN_SECONDS = 30
+    WAKEUP_INTERVAL = 1
+
+    def __init__(self):
+        super().__init__()
+        self.logger = logging.getLogger(__name__)
+        self.state = FeedbackState.SCALING_UP
+        self.messageQueue = queue.Queue()
+        self.shared_client_states = {}
+        self.total_active_client_count = 0
+        self.total_client_count = 0
+        self.sleep_start_time = None
+        self.last_error_time = None
+        self.last_scaleup_time = None
+
+    def handle_state(self):
```

Comment:
To follow the existing pattern in other actor classes, move all the concrete methods below all message related methods.